tests/testthat/test-build-search-docs.R

```diff
@@ -1,14 +1,26 @@
 test_that("docsearch.json and sitemap.xml are valid", {
-  skip_if_not_installed("jsonlite")
 
   search <- test_path("assets/search-site")
   on.exit(clean_site(search))
 
-  expect_output(init_site(search))
+  expect_output(build_site(search, new_process = FALSE))
   json <- path(search, "docs", "docsearch.json")
   expect_true(jsonlite::validate(read_lines(json)))
 
   xml <- path(search, "docs", "sitemap.xml")
   schema <- xml2::read_xml(path(search, "sitemaps-schema-0.9.xsd"))
   expect_true(xml2::xml_validate(xml2::read_xml(xml), schema))
 })
+
+test_that("build_search() builds the expected search.json", {
+  path <- test_path("assets/news")
+  pkg <- as_pkgdown(path, list(news = list(cran_dates = FALSE)))
+  pkg$bs_version <- 4
+  tmp <- withr::local_tempdir()
+  pkg$dst_path <- tmp
+  build_news(pkg)
+  build_home(pkg)
+  build_sitemap(pkg)
+  build_search(pkg)
+  expect_snapshot_file(path(tmp, "search.json"))
```

Comment:
I wonder whether in the tests I should re-save it as pretty JSON whose diff would be much easier to look at.
Or should I had a different behavior in pkgdown code itself depending on whether the code is run from?

Reply:
Could we test the search index _before_ it's jsonified?